app/src/main/java/com/humberto/tasky/agenda/presentation/agenda_details/AgendaDetailsScreen.kt

```diff
@@ -0,0 +1,148 @@
+package com.humberto.tasky.agenda.presentation.agenda_details
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.humberto.tasky.R
+import com.humberto.tasky.agenda.presentation.AgendaItemType
+import com.humberto.tasky.core.presentation.designsystem.TaskyTheme
+import com.humberto.tasky.core.presentation.designsystem.components.TaskyScaffold
+import com.humberto.tasky.core.presentation.designsystem.components.TaskyToolbar
+import com.humberto.tasky.core.presentation.ui.toFormattedUppercaseDateTime
+import com.humberto.tasky.main.navigation.AgendaDetails
+import java.time.ZonedDateTime
+
+@Composable
+fun AgendaDetailsScreenRoot(
+    agendaDetails: AgendaDetails,
+    onBackClick: () -> Unit,
+    viewModel: AgendaDetailsViewModel = hiltViewModel()
+) {
+    LaunchedEffect(agendaDetails) {
+        viewModel.setStateFromNav(agendaDetails)
+    }
+    val state by viewModel.agendaDetailsState.collectAsState()
```

Comment:
I'd recommend to use `collectAsStateWithLifecycle()` that since it's the safest way to collect flows in Android. For that, you'll need this dependency:
```
implementation "androidx.lifecycle:lifecycle-runtime-compose:2.x.x"
```
Then use it like this:
```
val state by viewModel.state.collectAsStateWithLifecycle()
```
If you have a simple flow it shouldn't make a difference, but as soon as you convert a normal `Flow` into a `StateFlow` with `stateIn`, the Flow operators will be executed even when the app is in the background (which you typically don't want when dealing with UI updates).